### DISCUSSION TITLE
GitHub Copilot in a Codespace authenticated with separate accounts

### DISCUSSION THREAD

**Discussion by bgolubovic** [community/community#48027] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Hi,

I have GitHub Copilot on my personal GitHub account. I am using my biz account to login to GitHub Codespaces with VS Code.
In VS Code I am logged in with booth accounts. I installed GitHub Copilot extension locally and in Codespaces but it does not work. 

It would be great if this is possible.

**Comment by KapitanOczywisty** [comment]
Copilot in codespaces for some reason forces use of the same account, ignoring vscode settings. I've modified `extension.js` to patch this:
```sh
sed -i 's/"true"===process\.env\.CODESPACES/false/' ~/.vscode-remote/extensions/github.copilot-*/dist/extension.js
```
Then `> Developer: Reload Window` - this works as long the codespace is running. Could be possible to run this in [`postAttachCommand`](https://containers.dev/implementors/json_reference/#lifecycle-scripts) or [dotfiles](https://docs.github.com/en/codespaces/customizing-your-codespace/personalizing-github-codespaces-for-your-account#dotfiles) though I've not tested any of these.

Still, this is only hack and I hope this will be fixed.
